packages/prerender-fargate/lib/prerender/server.js

```diff
@@ -36,8 +36,114 @@ server.use({
+            requestReceived: function(req, res, next) {
+                if(req.method !== 'GET' && req.method !== 'HEAD') {
+                    console.log("skipping requestReceived from S3 Cache... ")
+                    return next();
+                }
+
+                var key = req.prerender.url;
+
+                if (process.env.S3_PREFIX_KEY) {
+                    key = process.env.S3_PREFIX_KEY + '/' + key;
+                }
+
+                s3.getObject({
+                        Key: key
+                }, function (err, result) {
+
+                    if (!err && result) {
+                        console.log(result.Metadata);
+                        console.log("Found cached object: " + key);
+                        if (result.Metadata.location){
+                            res.setHeader('Location', result.Metadata.location);
+                        }
+                        return res.send(result.Metadata.httpreturncode, result.Body);
+                    } else {
+                        console.error(err);
+                    }
+
+                    next();
+                });
+            },
+
+            pageLoaded: function(req, res, next) {
+                const statusCodesToCache = ['200', '301', '302'];
+                let metaTagStatusCode = 200;
+                let location = '';
+
+                if (req.prerender.content && req.prerender.renderType == 'html') {
+                    let statusMatch = /<meta[^<>]*(?:name=['"]prerender-status-code['"][^<>]*content=['"]([0-9]{3})['"]|content=['"]([0-9]{3})['"][^<>]*name=['"]prerender-status-code['"])[^<>]*>/i,
+                    headerMatch = /<meta[^<>]*(?:name=['"]prerender-header['"][^<>]*content=['"]([^'"]*?): ?([^'"]*?)['"]|content=['"]([^'"]*?): ?([^'"]*?)['"][^<>]*name=['"]prerender-header['"])[^<>]*>/gi,
+                    head = req.prerender.content.toString().split('</head>', 1).pop(),
+                    // statusCode = 200,
+                    match;
+
+                    if (match = statusMatch.exec(head)) {
+                        metaTagStatusCode = match[1] || match[2];
+                        req.prerender.content = req.prerender.content.toString().replace(match[0], '');
+                        console.log("metaTagStatusCode: " + metaTagStatusCode);
+                    }
+
+                    while (match = headerMatch.exec(head)) {
+                        location = he.decode(match[2] || match[4]);
+                        res.setHeader(match[1] || match[3], location);
+                        req.prerender.content = req.prerender.content.toString().replace(match[0], '');
+                    }
+
+                    // Skip caching for the http response codes not in the list, such as 404
+                    if ( ! statusCodesToCache.includes(metaTagStatusCode.toString()) ) {
+                        console.log("metaTagStatusCode " + metaTagStatusCode + " is not in the cachable code list. Returning without caching.");
```

Comment:
Probably not a useful log either. Also fun JS syntax fact, it's generally preferred to use string literals rather than `+`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

This would become: 
```
console.log(`metaTagStatusCode ${metaTagStatusCode} is not in the cachable code list. Returning without caching.`);
```